pyscript.core/esm/plugins.js

```diff
@@ -40,16 +43,68 @@ export const handlePlugin = (node) => {
                     config,
                 );
                 engine.then((runtime) => {
-                    const module = registry.get(type);
-                    onRuntimeReady(node, {
+                    const module = create(registry.get(type));
+                    const {
+                        onBeforeRun,
+                        onBeforeRunAsync,
+                        onAfterRun,
+                        onAfterRunAsync,
+                        onRuntimeReady,
+                        codeBeforeRunWorker,
+                        codeBeforeRunWorkerAsync,
+                        codeAfterRunWorker,
+                        codeAfterRunWorkerAsync,
+                    } = options;
+
+                    // patch sync
+                    for (const [name, [before, after]] of [
+                        ["run", [onBeforeRun, onAfterRun]],
```

Comment:
Just a small question, how does this one work? Is it `name` has an array that may contain `before` and `after`?

Reply:
the signature is borrowed from `new Map([[a, b], [c, d]])` and in this case is more like `new Map([[a, [b, c]], [d, [e, f]]])` ... the grouping is after a name entry and one or more related methods for that name (precisely  in this case).

Reply:
That makes sense thanks!

Reply:
I've ditched `new Map` in front as that's all static code no *Map* runtime should strictly analize as I don't expect duplicated keys in the array 'cause I am dictating what the array should contain ... I hope this is reasonable, but for a one-off operation I wouldn't mind bringing in `new Map(...)` to implicitly describe the loop intent ... it felt YAGNI to me, but that's also just my opinion. Maybe a comment would help understanding the desired result?

Reply:
Yeah I think a comment would be good if you are happy  with it 😄

Reply:
Done.

Reply:
thank you, looks good!